better_budget.py

```diff
@@ -36,19 +39,27 @@ def compare_budget(dataframes, budget, total):
 			print('Oops! In your {} budget, you have spent ${:.2f}, which is ${:.2f} over budget.'.format(key, spent, abs(saved)))
 	print('there is {} unaccounted for.'.format(total-total_spent))
 
-def plot_budget(dataframes, budget_dict):
+def plot_budget(dataframes, budget_dict, keyword_dict):
 	cat_totals = {k: abs(v['Amount'].sum()) for k, v in dataframes.items()}
 	cat_labels = [str(k) for k in dataframes]
 
 	fig, ax = plt.subplots()
+        # plot budgeted amounts
 	plt.bar(cat_labels, [budget_dict[l] for l in cat_labels], alpha=0.6)
-	plt.bar(cat_labels, [cat_totals[l] for l in cat_labels], alpha=0.6)
+        # plot spent amounts, broken down first by category then by keyword
+        colors = ['C{}'.format(i) for i in range(1, 10)]
+        for cat in keyword_dict:
+            kw_amounts = amounts_by_keyword(dataframes[cat], keyword_dict[cat])
+            sorted_keys = sorted(kw_amounts, key=lambda x: kw_amounts[x], reverse=True)
+            for i, k in enumerate(sorted_keys):
+                bottom = sum(sorted(kw_amounts.values(), reverse=True)[:i])
+                plt.bar(cat, kw_amounts[k], bottom=bottom, color=colors[i], alpha=0.6)
 	plt.show()
 
 
 
 keywords = {
-	'fun': ['soc', 'puff', 'diner', 'beer'], 
+	'fun': ['soc', 'puff', 'diner', 'beer'],
```

Comment:
NIT: this should be a separate commit. It has nothing to do with the added functionality.

Ideally these edits should be rebased in their added commit, but since that commit is already merged to master thats not really an option